app/src/main/java/de/westnordost/streetcomplete/quests/max_height/Height.java

```diff
@@ -34,24 +41,43 @@ public Unit getUnit()
 
 	public boolean isEmpty()
 	{
-		return integerPart == -1 && fractionalPart == -1;
+		if (unit.equals(Unit.METRIC))
+		{
+			return meters == -1;
+		}
+		else
+		{
+			return feet == -1 && inches == -1;
+		}
 	}
 
 	public String toString()
 	{
 		if (unit.equals(Unit.METRIC))
 		{
-			return String.valueOf(integerPart) + "." + String.valueOf(fractionalPart);
+			return String.valueOf(meters);
 		}
 		else
 		{
 			//this adds an apostrophe and a double-quote to be in a format like e.g. 6'7"
-			return String.valueOf(integerPart) + "'" + String.valueOf(fractionalPart) + "\"";
+			return String.valueOf(feet) + "'" + String.valueOf(inches) + "\"";
+		}
+	}
+
+	public double getInMeters()
+	{
+		if (unit.equals(Unit.METRIC))
+		{
+			return meters;
+		}
+		else
+		{
+			return feetToMeter(Double.parseDouble(feet + "." + inches));
```

Comment:
How about something like:
`return ((Integer.parseInt(feet) * 12 + Integer.parseInt(inches)) * 0.0254);`

This would do three things:

* Correctly interpret inches as a division of 12 instead of 10.
* Correctly return double precision (0.0254 is exact, 3.2808 is approximate and barely enough for float let alone double).
* Perform a multiply rather than a divide (less expensive).

(Note: untested; I don't know if more text ("Double") is required. I also don't know why Double is preferred over Float; is Double somehow quicker than Float?)